internal/netsync/manager.go

```diff
@@ -763,6 +763,19 @@ func (m *SyncManager) processBlock(block *dcrutil.Block) (int64, error) {
 	if err != nil {
 		return 0, err
 	}
+
+	// Update the sync height when the block is higher than the currently best
+	// known value and it extends the main chain.
+	onMainChain := forkLen == 0
+	if onMainChain {
+		m.syncHeightMtx.Lock()
+		blockHeight := int64(block.MsgBlock().Header.Height)
+		if blockHeight > m.syncHeight {
```

Comment:
Should it be `!=`?  Although it's not normal for the mainchain to become something at a lower height than previous, but that's a possibility (e.g. this PR or working with the invalidateblock RPC etc.)

Reply:
That would kind of defeat the idea of the sync height which is to report the overall height it's syncing to.  Making this != would cause the height to (incorrectly) be updated as every new block is processed and connected to the main chain.

Reply:
I should also note that the sync height is set back to earlier heights in the case an invalid header is processed from the sync peer elsewhere, so it will be updated in the scenario you mentioned where a block is invalidated and new headers are discovered.

Reply:
> so it will be updated in the scenario you mentioned where a block is invalidated and new headers are discovered

Gotcha.  That's clear when I zoom out.